src/clj/rems/workflow/dynamic.clj

```diff
@@ -330,3 +357,58 @@
                           [{:type ::add-member :actor "applicant" :member "member1"}
                            {:type ::add-member :actor "applicant" :member "member2"}]
                           {:valid-user? #{"member1" "member2"}})))))
+
+(deftest test-possible-commands
+  (let [base {:applicantuserid "applicant"
+              :workflow {:type :workflow/dynamic
+                         :handlers ["assistant"]}}]
+    (testing "draft"
+      (let [draft (assoc base :state ::draft)]
+      (is (= #{::submit ::add-member}
+             (possible-commands "applicant" draft)))
+      (is (= #{}
+             (possible-commands "assistant" draft)))
+      (is (= #{}
+             (possible-commands "somebody else" draft)))))
+    (testing "submitted"
+      (let [submitted (assoc base :state ::submitted)]
+        (is (= #{::add-member}
+               (possible-commands "applicant" submitted)))
+        (is (= #{::approve ::reject ::return ::request-decision}
+               (possible-commands "assistant" submitted)))
+        (is (= #{}
+               (possible-commands "somebody else" submitted)))))
+    (testing "decision requested"
+      (let [requested (assoc base
+                             :state ::submitted
+                             :decider "decider")]
```

Comment:
It should be more maintainable to write the tests in terms of events that have happened, instead of modifying the application's internal state directly, because the application is event-sourced.